build-tests/api-documenter-test/etc/yaml/api-documenter-test/docclass1.yml

```diff
@@ -14,22 +14,22 @@ items:
       - typeScript
     type: class
     extends:
-      - api-documenter-test.DocBaseClass
+      - 'api-documenter-test!DocBaseClass:class'
     implements:
-      - api-documenter-test.IDocInterface1
-      - api-documenter-test.IDocInterface2
-    package: api-documenter-test
+      - 'api-documenter-test!IDocInterface1:interface'
+      - 'api-documenter-test!IDocInterface2:interface'
+    package: api-documenter-test!
     children:
-      - api-documenter-test.DocClass1.deprecatedExample
-      - api-documenter-test.DocClass1.exampleFunction
-      - api-documenter-test.DocClass1.exampleFunction_1
-      - api-documenter-test.DocClass1.interestingEdgeCases
-      - api-documenter-test.DocClass1.malformedEvent
-      - api-documenter-test.DocClass1.modifiedEvent
-      - api-documenter-test.DocClass1.regularProperty
-      - api-documenter-test.DocClass1.sumWithExample
-      - api-documenter-test.DocClass1.tableExample
-  - uid: api-documenter-test.DocClass1.deprecatedExample
+      - 'api-documenter-test!DocClass1#deprecatedExample:member(1)'
+      - 'api-documenter-test!DocClass1#exampleFunction:member(1)'
+      - 'api-documenter-test!DocClass1#exampleFunction:member(2)'
+      - 'api-documenter-test!DocClass1#interestingEdgeCases:member(1)'

```

Comment:
Even though they can't collide, looking at this kind of makes me want to have `:property`, `:accessor`, `:method`, and `:enummember`, but this is probably fine.

Reply:
We want this to be the same notation that humans use when creating `{@link}` tags, though.  Inexperienced developers may have trouble understanding the distinction between an `:accessor` versus a `:property`, and they may wonder why an `:enummember` is treated specially.

Most of the selector names are easy to learn because they correspond to a TypeScript keyword that introduces them (e.g. `constructor`, `var`, `function`, etc).  Thus people can learn that `:member` is the catch-all for constructs that don't have a natural TypeScript keyword.  (The `:index` and `:call` selectors are the only exceptions to this rule, but they will be rarely encountered, and somewhat special as they don't have an identifier.)